code/game/objects/items/stacks/golem_food/golem_status_effects.dm

```diff
@@ -73,28 +78,79 @@
 		to_chat(owner, span_notice(applied_fluff))
 	if (!overlay_state_prefix || !iscarbon(owner))
 		return TRUE
-	var/mob/living/carbon/golem_owner = owner
-	for (var/obj/item/bodypart/part in golem_owner.bodyparts)
+	// Make sure to set our human_owner
+	human_owner = owner
+	for (var/obj/item/bodypart/part in human_owner.bodyparts)
 		if (part.limb_id != SPECIES_GOLEM)
 			continue
 		var/datum/bodypart_overlay/simple/golem_overlay/overlay = new()
 		overlay.add_to_bodypart(overlay_state_prefix, part)
 		active_overlays += overlay
-	golem_owner.update_body_parts()
+	human_owner.update_body_parts()
+	// listen for species changes, for effects that immediately apply to golems or wouldn't apply to humans
+	RegisterSignal(human_owner, COMSIG_SPECIES_GAIN, PROC_REF(species_handling))
 	return TRUE
 
+/datum/status_effect/golem/proc/species_handling(datum/source, datum/species/new_species, datum/species/old_species)
+	SIGNAL_HANDLER
+	SHOULD_CALL_PARENT(TRUE)
+
+	return
+
+/datum/status_effect/golem/tick(seconds_between_ticks)
+	. = ..()
+	var/atom/movable/screen/alert/status_effect/golem_status/status_alert = linked_alert
+	if(!status_alert)
+		src.tick_interval = STATUS_EFFECT_NO_TICK
+		log_admin("[src] status effect has no alert, stopping tick.")
```

Comment:
Is this a debug log you forgot to remove? I don't think admins would care about this.